liquidctl/driver/smart_device.py

```diff
@@ -244,28 +247,52 @@ def __init__(self, device, description, speed_channel_count, color_channel_count
                           for i in range(color_channel_count)}
         super().__init__(device, description, speed_channels, color_channels, **kwargs)
 
-    def initialize(self, **kwargs):
-        """Initialize the device.
+    def initialize(self, direct_access=False, **kwargs):
+        """Initialize the device and the driver.
 
-        Detects all connected fans and LED accessories, and allows subsequent
-        calls to get_status.
+        Connected fans and LED accessories are detected.
+
+        This method should be called every time the systems boots, resumes from
+        a suspended state, or if the device has just been (re)connected.  In
+        those scenarios, no other method, except `connect()` or `disconnect()`,
+        should be called until the device and driver has been (re-)initialized.
+
+        Returns None or a list of `(property, value, unit)` tuples, similarly
+        to `get_status()`.
         """
 
-        self._write([0x1, 0x5c])  # initialize/detect connected devices and their type
-        self._write([0x1, 0x5d])  # start reporting
+        if self._hwmon and not direct_access:
+            _LOGGER.info('bound to %s kernel driver, assuming it is already initialized',
+                         self._hwmon.module)
+        else:
+            if self._hwmon:
+                _LOGGER.warning('forcing re-initialization despite %s kernel driver',
+                                self._hwmon.module)
+            self._write([0x1, 0x5c])  # initialize/detect connected devices and their type
+            self._write([0x1, 0x5d])  # start reporting
+            self.device.clear_enqueued_reports()
 
-    def get_status(self, **kwargs):
-        """Get a status report.
+        msg = self.device.read(self._READ_LENGTH)
 
-        Returns a list of (key, value, unit) tuples.
-        """
+        fw = '{}.{}.{}'.format(msg[0xb], msg[0xc] << 8 | msg[0xd], msg[0xe])
+        ret = [('Firmware version', fw, '')]
+
+        if self._color_channels:
+            lcount = msg[0x11]
+            ret.append(('LED accessories', lcount, ''))
+            if lcount > 0:
+                ltype, lsize = [('HUE+ Strip', 10), ('Aer RGB', 8)][msg[0x10] >> 3]
+                ret.append(('LED accessory type', ltype, ''))
```

Comment:
Should these string keys in this function be extracted into defined constants for consistency with the other driver updates?

Reply:
I only did it in places where there was no need to use a template, mostly because that would make the lines a bit more complex.  This obviously isn't the case for this particular string, but this and other devices have many fans, and use templates like `Fan {num} speed`; those would become `_STATUS_FAN_SPEED.format(num)`, which IMO is a bit less clear and a bit more verbose.

Additionally, since these strings are important keys, I feel abstracting away the key using constants is a tradeoff between lowering the noise when we're debugging how the values are computed versus adding an extra indirection when thinking revising which key/values a particular driver should report.